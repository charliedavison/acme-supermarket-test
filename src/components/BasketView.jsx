import React from 'react';
import PropTypes from 'prop-types';

class BasketView extends React.Component {
    constructor(props) {
        super(props);

        this.renderItems.bind(this);
    };

    renderItems() {
        return Object.keys(this.props.items).map(item => {
            return (
                <tr key={`basketItem_${item}`}>
                    <td>{this.props.items[item].name}</td>
                    <td>{this.props.items[item].quantity}</td>
                    <td>{`Â£${this.props.items[item].price}`}</td>
                </tr>
            )
        });
    };

    render() {
        return (
            <table>
                <thead>
                    <tr>
                        <th>{'Item Description'}</th>
                        <th>{'Quantity'}</th>
                        <th>{'Price'}</th>
                    </tr>
                </thead>
                <tbody>
                    {this.renderItems()}
                </tbody>
            </table>
        );
    }
};
BasketView.defaultProps = {
    items: {}
};
BasketView.propTypes = {
    items: PropTypes.object.isRequired
};
module.exports = BasketView;
