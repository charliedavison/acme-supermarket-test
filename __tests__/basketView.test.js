import React from 'react';
import { shallow } from 'enzyme';
import BasketView from './../src/components/BasketView.jsx';

describe('<BasketView />', () => {
    let component;
    
    afterEach(() => {
        component = null;
    });

    describe('No Items', () => {

        beforeEach(() => {
            component = shallow(<BasketView />);
        });

        test('Should match snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('Has Items', () => {

        const props = {
            items: {
                "CF1": { "quantity": 1, "productCode": "CF1", "name": "Coffee", "price": 11.23, "image": "coffee.jpg" }
            }
        };

        beforeEach(() => {
            component = shallow(<BasketView {...props} />);
        });

        test('Should match snapshot', () => {
            expect(component).toMatchSnapshot()
        });

    });
});