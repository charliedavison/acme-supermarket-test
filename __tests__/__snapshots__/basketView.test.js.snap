// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketView /> Has Items Should match snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Item Description
      </th>
      <th>
        Quantity
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="basketItem_CF1"
    >
      <td>
        Coffee
      </td>
      <td>
        1
      </td>
      <td>
        Â£11.23
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<BasketView /> No Items Should match snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Item Description
      </th>
      <th>
        Quantity
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
